source/Octopus.Tentacle.Tests.Integration/Support/Server.cs

```diff
@@ -1,22 +1,39 @@
 ﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
 using Halibut;
+using Serilog;
+using Octopus.Tentacle.Client.Retries;
 
 namespace Octopus.Tentacle.Tests.Integration.Support
 {
-    public class Server : IDisposable
+    public class Server : IAsyncDisposable
     {
+        private readonly ILogger logger;
         public IHalibutRuntime ServerHalibutRuntime { get; }
         public int ServerListeningPort { get; }
 
-        public Server(IHalibutRuntime serverHalibutRuntime, int serverListeningPort)
+        public Server(IHalibutRuntime serverHalibutRuntime, int serverListeningPort, ILogger logger)
         {
+            this.logger = logger.ForContext<Server>();
             this.ServerHalibutRuntime = serverHalibutRuntime;
             ServerListeningPort = serverListeningPort;
         }
 
-        public void Dispose()
+        public async ValueTask DisposeAsync()
         {
-            ServerHalibutRuntime.Dispose();
+            using var disposeCancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(10));
+            var cancellationToken = disposeCancellationTokenSource.Token;
+
+            logger.Information("Starting ServerHalibutRuntime.DisposeAsync");
+            var disposeTask = Task.Run(async () => await ServerHalibutRuntime.DisposeAsync(), CancellationToken.None);
```

Comment:
Just wondering, what's the difference between `Task.Run(async () => await ServerHalibutRuntime.DisposeAsync(), CancellationToken.None)` or `ServerHalibutRuntime.DisposeAsync().AsTask()`?

I saw `AsTask` used above, and wasn't sure what the difference here was.

Reply:
I was seeing `ServerHalibutRuntime.DisposeAsync().AsTask()` block and not be abandoned after the 10 second cancellation timeout. Wrapping in an explicit Task.Run to try and get it to behave as expected. I suspect something todo with SyncHalibut DisposeAsync not really being async and blocking.

Reply:
Ah yes, that makes sense. Do you think this would happen in the above places we use `AsTask` in the dispose method?

Reply:
Yeh it might not be quite right and there may still be some blocking but I'll revisit if we see that happening. With all the logging in Disposal, it's really easy to see where stuff is getting stuck now so Whac-A-Mole in this area should be a lot easier.